homework-submission/week4/src/stores/TodoList.js

```diff
@@ -0,0 +1,100 @@
+import { observable, action } from 'mobx';
+import TodoItems from '../sources/todoItems.json';
+
+class Store {
+  @observable
+  items = localStorage.items !== undefined
+    ? JSON.parse(localStorage.items)
+    : TodoItems;
+  @observable
+  newItem = {
+    newId:
+      localStorage.items !== undefined
+      ? JSON.parse(localStorage.items).length + 1
+      : TodoItems.length + 1,
+    newDescription: "",
+    newDeadline: "",
+  }
+
+  @action
+  checkAsDone = (id) => {
+    return this.items = this.items.map((item) =>
+      item[id] === id
+      ? {...item, done: !item.done }
+      : item
+    );
+  }
+
+  @action
+  addTodo = () => {
+    const newItem = {
+      id: this.newItem.newId,
+      description: this.newItem.newDescription,
+      deadline: this.newItem.newDeadline,
+      done: false,
+      editable: false
+    };
+
+    return this.items.push(newItem);
+
+  };
+
+  @action
+  resetNewDescription = () => {
+    return this.newItem.newDescription = "";
+  }
+  
+  @action
+  resetNewDeadline = () => {
+    return this.newItem.newDeadline = "";
+  }
+
+  @action
+  addNewDescription = (newDescription) => {
+    return this.newItem.newDescription = newDescription;
+  };
+
+  @action
+  addNewDeadline = (value) => {
+    return this.newItem.newDeadline = value;
+  };
+
+  @action 
+  removeTodo = (id) => {
+    id = parseInt(id, 10);
+    return this.items = this.items.filter(item => item.id !== id);
+  };
+
+  @action 
+  editTodo = (id) => {
+    id = parseInt(id, 10);
+    return this.items = this.items.map((item) =>
+      item[id] === id
```

Comment:
Here, you are accessing the object property using the `item[id]` way but this means you are accessing `item.1` is this possible? in this case you need to access the id property like this `item.id`

Reply:
Yes of course. I changed the way I access id but I didn't pay attention to update that and I kept reading it as id. 
Thank you for your time and review 🙏